tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SolutionEpicRepositoryTests.cs

```diff
@@ -0,0 +1,180 @@
+                Active = true
+            };
+        }
+
+        [SetUp]
+        public async Task Setup()
+        {
+            await Database.ClearAsync().ConfigureAwait(false);
+
+            await SupplierEntityBuilder.Create()
+                .WithId(SupplierId)
+                .Build()
+                .InsertAsync()
+                .ConfigureAwait(false);
+
+            await SolutionEntityBuilder.Create()
+                .WithId(Solution1Id)
+                .WithSupplierId(SupplierId)
+                .Build()
+                .InsertAsync()
+                .ConfigureAwait(false);
+
+            TestContext testContext = new TestContext();
+            _solutionEpicRepository = testContext.SolutionEpicRepository;
+        }
+
+        [Test]
+        public async Task UpdateSolutionWithOneEpicAsync()
+        {
+            await InsertCapabilityAsync(_capDetails[0]).ConfigureAwait(false);
+            await InsertSolutionCapabilityAsync(Solution1Id, _capDetails[0].Id).ConfigureAwait(false);
+            await InsertEpicAsync(_epicDetails[0]).ConfigureAwait(false);
+
+            var expectedEpic = new List<IClaimedEpicResult>
+            {
+                Mock.Of<IClaimedEpicResult>(e => e.EpicId == _epicDetails[0].Id && e.StatusName == StatusPassed)
+            };
+
+            await _solutionEpicRepository
+                .UpdateSolutionEpicAsync(Solution1Id,
+                    Mock.Of<IUpdateClaimedRequest>(c => c.ClaimedEpics == expectedEpic), new CancellationToken()).ConfigureAwait(false);
+
+            var solutionEpics = (await SolutionEpicEntity.FetchForSolutionAsync(Solution1Id).ConfigureAwait(false)).ToList();
+            solutionEpics.Count().Should().Be(1);
+            solutionEpics[0].Should().Be(_epicDetails[0].Id);
+        }
+
+        [Test]
+        public async Task UpdateSolutionWithMultipleEpicsAsync()
+        {
+            await InsertCapabilityAsync(_capDetails[0]).ConfigureAwait(false);
+            await InsertCapabilityAsync(_capDetails[1]).ConfigureAwait(false);
+            await InsertSolutionCapabilityAsync(Solution1Id, _capDetails[0].Id).ConfigureAwait(false);
+            await InsertSolutionCapabilityAsync(Solution1Id, _capDetails[1].Id).ConfigureAwait(false);
+
+            await InsertEpicAsync(_epicDetails[0]).ConfigureAwait(false);
+            await InsertEpicAsync(_epicDetails[1]).ConfigureAwait(false);
+
+            var expectedEpic = new List<IClaimedEpicResult>
+            {
+                Mock.Of<IClaimedEpicResult>(e => e.EpicId == _epicDetails[0].Id && e.StatusName == StatusPassed),
+                Mock.Of<IClaimedEpicResult>(e => e.EpicId == _epicDetails[1].Id && e.StatusName == StatusPassed)
+            };
+
+            await _solutionEpicRepository
+                .UpdateSolutionEpicAsync(Solution1Id,
+                    Mock.Of<IUpdateClaimedRequest>(c => c.ClaimedEpics == expectedEpic), new CancellationToken()).ConfigureAwait(false);
```

Comment:
Just noticed this interface name, could we change it to IUpdateClaimedEpicListRequest?